### DISCUSSION TITLE
Death/loss reason

### DISCUSSION THREAD

**Discussion by chillfox** [BattlesnakeOfficial/feedback#153] [Feature Requests]
The web interface shows a reason for why a snake died:
Out of health
Collided with itself
etc...

Please include this reason in the JSON object sent to /end and /move.
I would like to log the reasons my snake dies so that I see what is the most frequent cause of lost games and focus on that.

**Comment by NathanaelPadgett** [comment]
I'll second this feature request. This is how I phrased in it my tweet and on Discord: "Anyone know if there are plans for the BattleSnake API to tell the client in the END request the result of the game: "Collided with itself", "Out of health", "Lost head-to-head with XYZ", "Moved out of bounds" or at least Won/Draw/Eliminated values?"

**Reply by ItsGoldeneyes** [reply]
Consider it thirded. Very useful for data collection

**Reply by bvanvugt** [reply]
Yeah, this wouldn't be terribly hard to do, either. Currently we maintain various elimination codes inside the game engine, and then the UI builds that string dynamically.

We could push that string generation to the engine, which would allow for this.

**Reply by jasperstone** [reply]
Agreed. Cause of death would be really useful for automated testing and insights.
